### PR TITLE
:tada: Add support for new block types

### LINKED ISSUE #28: Widgets (slider, image edition)
A widget block that outputs a variable could be useful.

Examples:
- Slider block that outputs a float
- Image editor block where one could draw a number in MNIST-style and it could be instantly evaluated by a code block containing a classifier

### PR DESCRIPTION
**Pull request by vanyle**
This branch will resolve #28

### CODE REVIEW

File: opencodeblocks/graphics/blocks/block.py [hunk 1]
```diff
@@ -47,7 +46,8 @@ def __init__(self, block_type: str = 'base', source: str = '', position: tuple =
         QGraphicsItem.__init__(self, parent=parent)
         Serializable.__init__(self)
 
-        self.block_type = block_type
+        self.block_type = type(self).__name__
+        print("Type: ",self.block_type)
```
Comment (MathisFederico) [inline opencodeblocks/graphics/blocks/block.py:50]: Leftover print
```suggestion
```

**Review by MathisFederico — CHANGES_REQUESTED** [review]
Look good to me !

**Comment by vanyle** [comment]
Ne pas encore merge et review. Cette pull request est en cours d'écriture.
A la fin, il y aura toutes la partie qui lie le python au blocks ainsi que plus de types de blocs que simplement le slider.

**Comment by vanyle** [comment]
Next step : Fix the serialization issues and add another block type (the drawing thing).

The execution part probably won't be complete here as we rely too heavily on the execution flow feature that is not merged as of the time of writing.

**Comment by MathisFederico** [comment]
> Don't merge this just yet. This pull request is being worked on.
> 
> When it will be ready, it will feature more block types as well as the interactions between the python part and the slider part.

You can convert the PR to draft in that case

File: opencodeblocks/scene/clipboard.py [hunk 2]
```diff
@@ -80,6 +79,8 @@ def _find_bbox_center(self, blocks_data):
         return (xmin + xmax) / 2, (ymin + ymax) / 2
 
     def _deserializeData(self, data:OrderedDict, set_selected=True):
+        if data is None: return
```
Comment (MathisFederico) [inline opencodeblocks/scene/clipboard.py:82]: Why this ?

File: requirements-dev.txt [hunk 3]
```diff
@@ -7,4 +7,5 @@ pytest-qt
 pyautogui
 pylint>=2.12
 pylint-pytest
-autopep8
\ No newline at end of file
+autopep8
```
Comment (MathisFederico) [inline requirements-dev.txt:10]: Remove autopep8 -> replaced by black

**Review by MathisFederico — CHANGES_REQUESTED** [review]
A few changes, and lack of tests, but hey we gotta go fast for this 1st release !

File: opencodeblocks/scene/clipboard.py (hunk 2 above)
Comment (vanyle) [reply opencodeblocks/scene/clipboard.py:82]: Avoid crashes then the clipboard does not contain valid JSON data.

Comment (MathisFederico) [reply opencodeblocks/scene/clipboard.py:82]: Yeah ok whatever

**Review by MathisFederico — APPROVED** [review]

**Pull request merged**